core/src/mast/node/dyn_node.rs

```diff
@@ -57,9 +54,17 @@ impl DynNode {
 impl MastNodeErrorContext for DynNode {
```

Comment:
Do we actually need the `MastNodeErrorContext` trait? AFAICT, this is used primarily in `ErrorContextImpl` and I think we can pass just `MastNodeId` there to get a list of decorators related to a given operation. We would need to move the `get_assembly_op()` implementation into `MastForest` - but it shouldn't be too difficult. It could look something like:

```rust
impl MastForest {
    fn get_assembly_op(
        &self,
        node_id: MastNodeId,
        target_op_idx: Option<usize>,
    ) -> Option<AssemblyOp> {
        ...
    }
}
```